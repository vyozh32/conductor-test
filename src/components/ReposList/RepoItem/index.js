import React, { Component, Fragment } from 'react';

class RepoItem extends Component {
	constructor(props) {
		super(props);
		this.state = {
			isActive: false
		};
	}

	handleItem = () => {
		this.setState({
			isActive: !this.state.isActive
		});
	};

	render() {
		const { description, archived, created_at, fork, forks, name, open_issues, index } = this.props;
		return (
			<Fragment>
				<tr className="table-row" onClick={this.handleItem}>
					<td colspan="8" scope="col">
						{name}
					</td>
				</tr>
				{this.state.isActive ? (
					<tr>
						<td scope="col">{index + 1}</td>
						<td scope="col">{name}</td>
						<td scope="col">{description || '-'}</td>
						<td scope="col">{archived || '-'}</td>
						<td scope="col">{created_at}</td>
						<td scope="col">{fork}</td>
						<td scope="col">{forks || '0'}</td>
						<td scope="col">{open_issues || '0'}</td>
					</tr>
				) : null}
			</Fragment>
		);
	}
}

export default RepoItem;
