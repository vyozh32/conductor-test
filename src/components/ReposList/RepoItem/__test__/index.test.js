import React from 'react';
import RepoItem from '../index';
import renderer from 'react-test-renderer';

describe('<RootContainer>', function() {
	const props = {
		id: 1,
		name: 'Repos name',
		description: 'repos desc some test text',
		archived: 10,
		created_at: '322 228 1488 1337',
		fork: 10,
		forks: 100,
		name: 'Repos name',
		open_issues: 228,
		index: 10
	};

	it('renders correctly', function() {
		const component = renderer.create(<RepoItem {...props} />);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('renders with opened repo', function() {});
	const component = renderer.create(<RepoItem {...props} />);
	component.getInstance().handleItem();
	expect(component.getInstance().state.isActive).toEqual(true);
	expect(component.toJSON()).toMatchSnapshot();
});
