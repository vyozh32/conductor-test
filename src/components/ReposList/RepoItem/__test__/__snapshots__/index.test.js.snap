// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  <tr
    className="table-row"
    onClick={[Function]}
  >
    <td
      colspan="8"
      scope="col"
    >
      Repos name
    </td>
  </tr>,
  <tr>
    <td
      scope="col"
    >
      11
    </td>
    <td
      scope="col"
    >
      Repos name
    </td>
    <td
      scope="col"
    >
      repos desc some test text
    </td>
    <td
      scope="col"
    >
      10
    </td>
    <td
      scope="col"
    >
      322 228 1488 1337
    </td>
    <td
      scope="col"
    >
      10
    </td>
    <td
      scope="col"
    >
      100
    </td>
    <td
      scope="col"
    >
      228
    </td>
  </tr>,
]
`;

exports[`<RootContainer> renders correctly 1`] = `
<tr
  className="table-row"
  onClick={[Function]}
>
  <td
    colspan="8"
    scope="col"
  >
    Repos name
  </td>
</tr>
`;
